benchmark/functional/test_httpapi.py

```diff
@@ -0,0 +1,28 @@
+from testtools import TestCase
+
+from twisted.web.http import CREATED
+
+from ..httpapi import TxMongoBackend
+from ..test.test_httpapi import BenchmarkAPITestsMixin
+
+
+class TxMongoBenchmarkAPITests(BenchmarkAPITestsMixin, TestCase):
+    def setUp(self):
+        self.backend = TxMongoBackend()
+        self.addCleanup(self.backend.disconnect)
+        super(TxMongoBenchmarkAPITests, self).setUp()
+
+    def submit(self, result):
+        """
+        Submit a result.
+        """
+        req = super(TxMongoBenchmarkAPITests, self).submit(result)
+
+        def add_cleanup(response):
+            if response.code == CREATED:
+                location = response.headers.getRawHeaders(b'Location')[0]
+                self.addCleanup(lambda: self.agent.request("DELETE", location))
```

Comment:
Actually that's how I originally implemented it, but then I decided that it was safer to remove individual records just in case the tests would be run against a wrong MongoDB server by accident, or some such.
By I do not have a strong opinion.